consensus/src/main/java/org/apache/iotdb/consensus/ratis/SnapshotStorage.java

```diff
@@ -141,12 +97,17 @@ public SnapshotInfo getLatestSnapshot() {
     }
     TermIndex snapshotTermIndex = Utils.getTermIndexFromDir(latestSnapshotDir);
 
+    List<File> actualSnapshotFiles = applicationStateMachine.getSnapshotFiles(latestSnapshotDir);
+    if (actualSnapshotFiles == null) {
+      return null;
+    }
+
     List<FileInfo> fileInfos = new ArrayList<>();
-    for (Path file : getAllFilesUnder(latestSnapshotDir)) {
-      if (file.endsWith(".md5")) {
+    for (File file : actualSnapshotFiles) {
+      if (file.toPath().endsWith(".md5")) {
```

Comment:
The interface is in `IStateMachine`. I find that the other interfaces in `IStateMachine` all use `File` instead of `Path`. However, Ratis requires snapshot info to be passed as List of Paths. Should we change the interface in `IStateMachine` or we keep using an adapter here?